apt-hook/hook.cc

```diff
@@ -271,5 +277,32 @@ int main(int argc, char *argv[])
       ioprintf(std::cout, "\n");
    }
 
+
+   if (hook == "apt-install-pre-invoke") {
```

Comment:
might be able to parametrize this hook name and use it to dynamically source the /var/lib/ubuntu-advantage/<hook_name>-message file if present. So we might not need to touch this much if we want to make it a simple pass through for any custom apt-hook messaging based on the arg1 passed to apt-esm-hook

Reply:
We partially discussed this out-of-band yesterday - posting here for posterity.

This is a very good point! It might not be doable because this hook needs to internally calculate which template file to use in at least one circumstance (`apt upgrade` vs `apt dist-upgrade` messages). If it turns out it is doable, we should do it.